components/sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link.jsx

```diff
@@ -41,6 +41,30 @@ export default class SidebarChannelButtonOrLink extends React.PureComponent {
     constructor(props) {
         super(props);
         this.gmItemRef = React.createRef();
+        this.displayNameRef = React.createRef();
+    }
+
+    state = {
+        showTooltip: false,
+    }
+
+    componentDidMount() {
+        this.enableToolTipIfNeeded();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.displayName !== this.props.displayName) {
+            this.enableToolTipIfNeeded(true);
+        }
+    }
+
+    enableToolTipIfNeeded = (disableIfAlreadyEnabled = false) => {
+        const element = this.displayNameRef.current;
+        if (element && element.offsetWidth < element.scrollWidth) {
+            this.setState({showTooltip: true});
+        } else if (disableIfAlreadyEnabled) {
```

Comment:
Is this flag necessary? Should it be fine if it always turns if the above condition is not met?

Reply:
@marianunez, good point!

I wanted to avoid potentially unneeded re-renders when the `enableToolTipIfNeeded` is called in 'componentDidMount', however, I'd missed that the component is actually a `PureComponent`. So, I will remove the flag and always set to `false` if the condition is not met